### DISCUSSION TITLE
Can't figure out the syntax for Optional[T] to use with dataclasses

### DISCUSSION THREAD

**Discussion by vidjuheffex** [hylang/hy#2536] [Q&A] [answered]
From what I gather this looks correct:

```clojure
(defclass [dataclass] MyClass []
  "A dataclass to hold round information"
  #^ (of Optional int) myattr None)
  ```
  
  If I evaluate that it evaluates with no issue
  
  if I make `(MyClass)` it complains about the positional argument myattr being missing, but it's optional and has an Assignment?

**Comment by Kodiologist** [comment]
What would be the Python equivalent of the desired Hy code?

**Reply by vidjuheffex** [reply]
```python
from dataclasses import dataclass
from typing import Optional

@dataclass
class MyClass:
    myattr: Optional[int] = None
```    
  so MyClass can be made with simply `(MyClass)`

**Reply by Kodiologist** [reply]
In that case, Sunjay's answer should have you covered.

**Comment by scauligi** [comment, marked as answer]
`hy --spy` (or `hy2py`) is your friend!
The `#^` macro doesn't generate an assignment, it's just the annotation; you still need a `setv`. So your current Hy code declares a field `myattr` with no assignment, followed by a standalone expression `None`:

```python
# => (defclass [dataclass] MyClass []
# ...   "A dataclass to hold round information"
# ...   #^ (of Optional int) myattr None)
@dataclass
class MyClass:
    """A dataclass to hold round information"""
    myattr: Optional[int]
    None
```

You probably want this:
```hy
(defclass [dataclass] MyClass []
  (setv #^ (of Optional int) myattr None))
```

which translates to this:
```python
@dataclass
class MyClass:
    myattr: Optional[int] = None
```

**Reply by vidjuheffex** [reply]
Oh wow, thank you!!
